src/Core/Config/ReaddressableConfigInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of Mannequin.
+ *
+ * (c) 2017 Last Call Media, Rob Bayliss <rob@lastcallmedia.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace LastCall\Mannequin\Core\Config;
+
+/**
+ * Defines a configuration object that can be reloaded inside a new PHP process.
+ *
+ * The two pieces we need to reload are:
+ *   - The autoload.php file that was used in initially loading the object.
+ *   - The file that causes the instantiation of the object.
+ *
+ * In most cases, we don't actually need our config to be readdressable, but
+ * for the development server, we do.  In all other instances, we should be
+ * type hinting for ConfigInterface, not ReaddressableConfigInterface.
+ */
+interface ReaddressableConfigInterface extends ConfigInterface
```

Comment:
The name on this class really threw me off for awhile; maybe I'm just not familiar with the term "readdressable", but I kept feeling like these configs were going to be moving around the filesystem or something. I can't think of any names I like better, though.